### PR TITLE
Math magicians: tests

### PR DESCRIPTION
**Pull request by a-laarabi**
# What?
> In this repository, me @a-laarabi  and my partner @nati2323 created files to test our App.
### Highlights:
- We did make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- We used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- We documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- We followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
- We did import the calculator and use it in the main component (App.js).
- We test it in the browser; it render without problems.

## Screenshot for our testing files
Capture d’écran 2022-08-19 à 19 28 24

Cheer and happy reviewing 🥇

### CODE REVIEW

File: src/test/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import operate from '../logic/operate';
+
+describe('test the logic of the application', () => {
+  it('take two numbers and return the operation of them', () => {
+    expect(operate(2, 5, '+')).toBe('7');
+    expect(operate(2, 5, '-')).toBe('-3');
+    expect(operate(2, 5, 'x')).toBe('10');
+  });
+});
```
Comment (KDlamini) [inline src/test/operate.test.js]: Kindly test all arithmetic operations. You are missing tests for division (÷) and modulus (%).

- Please make sure that each test has its own description that explains the operation being tested, in your case you can have something like:

```
  it('Test for addition operation', () => {
    expect(operate(2, 5, '+')).toBe('7');
  });

  it('Test for subtraction operation', () => {
    expect(operate(2, 5, '-')).toBe('-3');
  });

... and so forth.
```

Another example:
```
describe('testing operations', () => {
  test('test addition', () => {
    const result = operate(5, 9, '+');
    expect(result).toBe('14');
  });

  test('test subtraction', () => {
    const result = operate(5, 9, '-');
    expect(result).toBe('-4');
  });

  test('test multiplication', () => {
    const result = operate(5, 9, 'x');
    expect(result).toBe('45');
  });

  test('test division', () => {
    const result = operate(5, 9, '÷');
    expect(result).toBe('0.55555555555555555556');
  });

  test('test modulus', () => {
    const result = operate(5, 9, '%');
    expect(result).toBe('5');
  });
});
```

File: src/test/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import operate from '../logic/operate';
+
```
Comment (KDlamini) [inline src/test/operate.test.js:2]: Kindly write tests for all operations with the `calculate.js` component as well.

For example:

```
import calculate from '../../logic/calculate';

describe('testing the calculate function ', () => {
  const obj = { total: null, next: null, operation: null };

  test('test addition', () => {
    let result = calculate(obj, '4');
    result = calculate(result, '+');
    result = calculate(result, '7');
    result = calculate(result, '=');

    expect(result.total).toBe('11');
  });

  test('test subtraction', () => {
    let result = calculate(obj, '8');
    result = calculate(result, '-');
    result = calculate(result, '3');
    result = calculate(result, '=');

    expect(result.total).toBe('5');
  });

... and so forth
end
```

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @a-laarabi, @nati2323 🙋‍♀️,  

Great work so far! 💯 
You still need to work on some issues to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 
✔️  Good PR title and description
✔️  Correct use of Gitflow 
✔️  Commit messages are meaningful
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — APPROVED** [review]
Hi @a-laarabi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**